app/recipes/routes.py

```diff
@@ -0,0 +1,454 @@
+#!/usr/bin/env python3
+import html
+import math
+import re
+from app import mongo
+from bson.objectid import ObjectId
+from datetime import datetime
+from flask import Blueprint, current_app, render_template, redirect, request, url_for, flash, session, Markup
+from slugify import slugify
+
+recipes = Blueprint("recipes", __name__)
+
+# database collection variables
+allergens_collection = mongo.db.allergens
+desserts_collection = mongo.db.desserts
+measurements_collection = mongo.db.measurements
+recipes_collection = mongo.db.recipes
+users_collection = mongo.db.users
+
+#----- Global Helper -----#
+def get_total_recipes():
+        return int(recipes_collection.count())
+@recipes.context_processor
+def total_recipes():
+        return dict(total_recipes=get_total_recipes)
+
+
+#---------- CRUD: CREATE | READ | UPDATE | DELETE ----------#
+
+# (Crud) ----- CREATE a new dessert -----#
+@recipes.route("/add")

```

Comment:
Use restful routes - https://learn.co/lessons/sinatra-restful-routes-readme
So this route should be `/desserts/new`

Reply:
Took a bit of time, but that's done now for all routes and templates according to the RESTful routes logic. Will be with the next refactoring push.